swap_meet/vendor.py

```diff
@@ -1,2 +1,68 @@
+from operator import attrgetter
+
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        '''Initializes a Vendor object with optional inventory argument'''
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        '''Takes in item object, appends it to inventory'''
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        '''Takes in item object, removes it from inventory''' 
+        if item not in self.inventory:
+            return False    
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        '''Input: category(string), returns inventory list with items of that category'''
+        return [item for item in self.inventory if item.category == category]
+        
+    def swap_items(self, other, my_item, their_item):
+        '''Inputs: other Vendor object, this vendor's item object, other vendor's item.
+        Swaps item objects between this vendor and other vendor's inventories'''
+
+        if my_item in self.inventory and their_item in other.inventory:
+            # item returned from remove() method gets used as the argument for add() method     
+            self.add(other.remove(their_item))
+            other.add(self.remove(my_item))
+            return True
+    
+    def swap_first_item(self, other):
+        '''Input: other vendor object. Swaps the first items in current and other vendor's inventories'''
+        
+        if self.inventory and other.inventory:  
+            return self.swap_items(other, self.inventory[0], other.inventory[0])
```

Comment:
I like that you've relegated the `swap_items()` helper function to return the right boolean here!